prow/rewriteConfig.go

```diff
@@ -0,0 +1,194 @@
+func countLeadingSpace(line string) int {
+	i := 0
+	for _, runeValue := range line {
+		if runeValue == ' ' {
+			i++
+		} else {
+			break
+		}
+	}
+	return i
+}
+
+// Add element to the slice at the indexed position and move all the strings from
+// the index one position to the right.
+func appendAtIndex(original []string, element string, index int) []string {
+	original = append(original, "")
+	copy(original[index+1:], original[index:])
+	original[index] = element
+
+	return original
+}
+
+// Copy the content of one slice to another.
+func copySlice(input []string) []string {
+	newSlice := []string{}
+	newSlice = append(newSlice, input...)
+
+	return newSlice
+}
+
+// Find index of master section in each branch split for the new branch and
+// get content from master section with mandatory "merges-blocked-needs-admin" part.
+func getContentOfNewBranch(branchLines []string, newBranch string) (int, string) {
+	if len(branchLines) < 2 {
+		return -1, ""
+	}
+	firstBranchLine := branchLines[1]
+	spacesForBranch := countLeadingSpace(firstBranchLine)
+	masterStart := -1
+	masterStop := -1
+	for j := 1; j < len(branchLines); j++ {
+		line := branchLines[j]
+		if len(line) > spacesForBranch-1 {
+			checkForMaster := line[spacesForBranch:]
+			if strings.Contains(checkForMaster, "master:") {
+				masterStart = j + 1
+			}
+		}
+
+		if masterStart != -1 && masterStop == -1 && j > masterStart {
+			if countLeadingSpace(line) <= spacesForBranch {
+				masterStop = j - 1
+			}
+		}
+	}
+
+	if masterStart != -1 && masterStop == -1 {
+		masterStop = len(branchLines) - 1
+	}
+
+	if masterStart != -1 && masterStop != -1 {
+
+		contextStart := -1
+		contextStop := -1
+		contextStrings := -1
+
+		secondBranchSpaces := countLeadingSpace(branchLines[masterStart])
+
+		for m := 0; m < masterStop-masterStart+1; m++ {
+			masterLine := branchLines[m+masterStart]
+			if strings.Contains(masterLine, "contexts:") {
+				contextStart = m + 1
+				contextStrings = countLeadingSpace(masterLine)
+			}
+
+			if contextStart != -1 && contextStrings != -1 && contextStop == -1 && m > contextStart {
+
+				if strings.Compare(string(masterLine[contextStrings]), "-") != 0 {
+					contextStop = m
+				}
+			}
+		}
+
+		if contextStart != -1 && contextStop == -1 {
+			contextStop = masterStop - masterStart + 1
+		}
+		containsNeedsAdmin := false
+		newBranchSlice := copySlice(branchLines[masterStart : masterStop+1])
+		if contextStart != -1 && contextStop != -1 {
+			for n := contextStart; n < contextStop; n++ {
+				contextLine := branchLines[n+masterStart]
+
+				if strings.Contains(contextLine, "merges-blocked-needs-admin") {
+					containsNeedsAdmin = true
+					break
+				}
+			}
+
+			if !containsNeedsAdmin {
+				numSpaces := countLeadingSpace(newBranchSlice[contextStart])
+				stringAdmin := strings.Repeat(" ", numSpaces) + "- \"merges-blocked-needs-admin\""
+
+				newBranchSlice = appendAtIndex(newBranchSlice, stringAdmin, contextStop-1)
+			}
+
+		} else {
```

Comment:
if it is already contains needs admin why we need to do all these additional steps?